src/containers/ModelPanel/index.tsx

```diff
@@ -79,10 +89,25 @@ const ModelPanel: React.FC<ModelPanelProps> = ({
     isNetworkedFile,
     changeToNetworkedFile: loadNetworkFile,
     recentColors,
-    applyUserColor,
     setRecentColors,
     selectedAgentMetadata,
+    setSelectedUIDisplayData,
+    setCurrentColorSetting,
+    storeDisplayDataInBrowser,
+    currentUIDisplayData,
 }): JSX.Element => {
+    const updateSelectedUiDisplayData = (colorChange: ColorChange) => {
+        const newUIData = applyColorChangeToUiDisplayData(
+            colorChange,
+            currentUIDisplayData
+        );
+        setSelectedUIDisplayData(newUIData);
+        setCurrentColorSetting({
+            currentColorSetting: ColorSetting.UserSelected,
+        });
+        storeDisplayDataInBrowser(newUIData);
```

Comment:
without digging into the details, but just looking at this, it seems like a lot of state setting to be happening in a component. I would move this into a logic.

Reply:
So this is work I specifically moved _out of logics_  in conversation with @frasercl about being more strict in the use of `redux-logic` for handling async operations only.

Can I request that we converge on an idea of what `logics` are for and then use them that way, because at this point I'll be changing things back and forth between different PR comment cycles.

Reply:
Checking the `state/README.md` says that `Logic` is for: "AJAX calls or complex business logic", and not strictly for async code.

@frasercl can you live with this philosophy?

Reply:
I would consider communicating with the browser storage as syncing with an external api

Reply:
I agree with the "AJAX calls" element of the description above but not necessarily with the "complex business logic" part. As I understand it (and once again, I'm not an authority, I'm trying to work all this out too!), part of redux's model is that actions don't necessarily take the form of "here's a specific element of state I want to set." They correspond to the real-world events that might require a state transformation - literally all the "actions" that a user or an external system are able to take. For instance, if clicking some button requires updating the enabled/disabled state of an agent _and_ setting its color, you don't necessarily want to dispatch a "set enabled" action _and_ a "set color" action, you might just want an "enabled/color change" action. So in general, if you're finding yourself dispatching a lot of actions and want to consolidate them, the tool to reach for is a more complex reducer, not necessarily a logic.

In this case though, where _both_ complex business logic and external API synchronization are happening, I think a logic makes sense.

Reply:
Thanks again for all the thought and consideration here.

After digging into the `redux-logic` docs a bit, I'm coming down on the side of "synchronous business logic is appropriate for the website's `logics`", because 
1) It keeps a lid on the number of redux related files where you have business logic to keep track of.
2) The `transform`/`validate` hook in `redux-logic` seems to make this use case explicit: validating/altering the payload in some way and then either proceeding, rejecting the action, or providing a different action to proceed with, so I'm going to try to make use of that.